src/backend/joanie/core/filters/client.py

```diff
@@ -79,3 +83,9 @@ def filter_has_listed_course_runs(self, queryset, _name, value):
             filtered_queryset = queryset.exclude(course_runs__is_listed=True)
 
         return filtered_queryset.distinct()
+
+    def filter_product_type(self, queryset, _name, value):
+        """
+        Filter resource by looking for products with matching type
```

Comment:
why don't we say "filter courses"? It will be clearer cause that's what we are filtering here:
```suggestion
        Filter courses by looking forbrelated products with matching type
```